keras_cv/layers/preprocessing/random_resized_crop_test.py

```diff
@@ -123,3 +123,28 @@ def test_crop_area_factor_errors(self, crop_area_factor):
                 aspect_ratio_factor=(3 / 4, 4 / 3),
                 crop_area_factor=crop_area_factor,
             )
+
+    def test_augment_with_segmentation_mask(self):
+        input_image_shape = (self.batch_size, self.height, self.width, 3)
+        mask_shape = (self.batch_size, self.height, self.width, 1)
+        image = tf.random.uniform(shape=input_image_shape, seed=self.seed)
+        mask = tf.cast(
+            tf.random.uniform(shape=mask_shape, seed=self.seed) > 0.5, tf.float32
+        )
+
+        input = {"images": image, "segmentation_masks": mask}
+
+        layer = preprocessing.RandomResizedCrop(
+            target_size=self.target_size,
+            aspect_ratio_factor=(3 / 4, 4 / 3),
+            crop_area_factor=(0.8, 1.0),
+            seed=self.seed,
+        )
+
+        input_mask_resized = tf.image.resize(mask, self.target_size)
+
+        output = layer(input, training=True)
+        self.assertNotAllClose(output["segmentation_masks"], input_mask_resized)
```

Comment:
Can we do more strict assertion, eg make sure we don't mix the class of the mask?

Reply:
I think this comment is based on an old diff -- the most recent version of this code does verify that we only output classes `0` and `7` using `self.assertAllInSet(output["segmentation_masks"], [0, 7])`